android/app/build.gradle

```diff
@@ -325,7 +325,7 @@ android {
 
   externalNativeBuild {
     cmake {
-      version '3.30.3+'
+      version '3.22.1+'
```

Comment:
What is the reason of downgrading?

Reply:
> Actually a new 3.30.3 cmake is not a hard requirement, the project builds just fine with older 3.22.1 (and prob even 3.18.1).

Its min required version, not a specific version OM depends on.

Reply:
The latest cmake works better with cmake profiles and Visual Studio for Windows. It will unblock these parts of the project.

Reply:
Do we use features of the new cmake in our CMakeFiles? No (right?)

Having a '3.22.1+' means any version newer than that is fine to be installed and people will reap all benefits of it.

At the same time people with e.g. older systems where newer cmake is not available, won't have to go through a trouble building it themselves meeting all the deps, etc.
(e.g. even 3.22.1 is not available for Ubuntu 20.04 and older and we had add special instructions where to get it; with 3.30.3+ even more distros will be excluded)

Reply:
E.g. we don't require Python 3.13 (the latest stable). We require Python 3.7+.

Reply:
I did want to have a more recent minimum version (3.24+), but not because of better integration with VS.
The reason was https://github.com/organicmaps/organicmaps/pull/7836#discussion_r1554618966

Please note that CMake 3.22.1 is 3 years old now. Ubuntu 22.04 (I believe this is why you selected this specific version) users can use [Kitware PPA](https://apt.kitware.com/) to get newer version of CMake.

Reply:
> I did want to have a more recent minimum version (3.24+), but not because of better integration with VS. The reason was [#7836 (comment)](https://github.com/organicmaps/organicmaps/pull/7836#discussion_r1554618966)
> 

Sure if you're going to use this feature (or any other) in one of your next PRs, then no point to downgrade in this PR (I'll remove this change).

But if its more like "maybe we'll use it some day in the future" - then let's bump only when its really needed.

> Please note that CMake 3.22.1 is 3 years old now. Ubuntu 22.04 (I believe this is why you selected this specific version) users can use [Kitware PPA](https://apt.kitware.com/) to get newer version of CMake.

Even the current 24.04 LTS users will have to use the PPA (as 24.04 has cmake 3.28.3). For the 2.33.1 cmake only 20.04 LTS users have to use the PPA.
As we don't use the newer cmake features yet, let's not add unjustified complexity to the dev env setup.

On a sidenote, it'd be good to agree upon some criteria how old dev tools and/or major distibutions we do try to support. E.g. if its "the oldest Ubuntu LTS" still maintained then its 20.04 which is supported till May 2025.

Reply:
`sudo snap install cmake` does it simply and easily on Ubuntu 22 and 24, no?

Reply:
> But if its more like "maybe we'll use it some day in the future" - then let's bump only when its really needed.

Yeah, it's more like that. I'm fine with keeping 3.22.1 for the time being. But it would be reasonable to also update it here:
https://github.com/organicmaps/organicmaps/blob/5eef1ef00d2aee1e92b3520c061eeb0e3aab8f4e/CMakeLists.txt#L1

Reply:
@Osyotr good catch!
Actually even 3.22.1 is not needed and 3.18 is sufficient still for our current use :)

Reply:
No, 3.18 doesn't work with Apple's ARM processors on Android.